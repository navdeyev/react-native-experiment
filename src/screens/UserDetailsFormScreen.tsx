import i18n from 'i18n-js';
import {omit} from 'ramda';
import * as React from 'react';

import {HasGetParam, User} from 'src/domains/types';
import ScreenLayout from 'src/components/ScreenLayout/ScreenLayout';
import UserDetailsForm from 'src/components/UserDetailsForm/UserDetailsForm';

import validate from 'src/domains/userDetailsForm/userDetailsFormService';

interface ScreenParams {
    user: User
}

export interface Props {
    navigation: HasGetParam<ScreenParams>
}

export const navigationOptions = ({navigation}: Props) => {
    const user = navigation.getParam('user');
    return {
        title: `${i18n.t('edit')}: ${user.username}`
    };
};

class UserDetailsFormScreen extends React.Component<Props> {
    static navigationOptions = navigationOptions;

    render() {
        const user: User = this.props.navigation.getParam('user');
        const initialValues = omit(['id', 'address', 'company'])(user);

        return (
            <ScreenLayout>
                <UserDetailsForm initialValues={initialValues} onSubmit={validate}/>
            </ScreenLayout>
        );
    }
}

export default UserDetailsFormScreen;
