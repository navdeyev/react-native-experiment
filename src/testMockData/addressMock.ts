import {Address, Geo} from 'src/domains/types';

export const geo: Geo = {
    lat: '-37.3159',
    lng: '81.1496'
};

export const address: Address = {
    street: 'Kulas Light',
    suite: 'Apt. 556',
    city: 'Gwenborough',
    zipcode: '92998-3874',
    geo
};
