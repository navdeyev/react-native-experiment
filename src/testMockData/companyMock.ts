import {Company} from 'src/domains/types';

export const company: Company = {
    name: 'Romaguera-Crona',
    catchPhrase: 'Multi-layered client-server neural-net',
    bs: 'harness real-time e-markets'
};
