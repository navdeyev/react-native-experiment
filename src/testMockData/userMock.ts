import {User} from '../domains/types';

import {address} from './addressMock';
import {company} from './companyMock';

export const user: User = {
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz',
    address,
    phone: '1-770-736-8031 x56442',
    website: 'hildegard.org',
    company
};
