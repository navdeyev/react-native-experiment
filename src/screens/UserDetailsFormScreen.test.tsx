import * as React from 'react';
import {shallow} from 'enzyme';

import UserDetailsFormScreen, {navigationOptions, Props} from './UserDetailsFormScreen';

import {user} from 'src/testMockData/userMock';

describe('UserDetailsFormScreen', () => {

    let props: Props;
    beforeEach(() => {
        props = {
            navigation: {
                getParam: jest.fn(() => user)
            }
        };
    });

    it('displays username in the navigation section', () => {
        expect(navigationOptions(props)).toEqual(expect.objectContaining({
            title: `Edit: ${user.username}`
        }));
    });

    it('renders a component', () => {
        const rendered = shallow(<UserDetailsFormScreen {...props} />);
        expect(rendered.getElement()).toMatchSnapshot();
    });
});