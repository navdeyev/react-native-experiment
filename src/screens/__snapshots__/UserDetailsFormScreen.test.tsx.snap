// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailsFormScreen renders a component 1`] = `
<Styled(Component)>
  <ReduxForm
    initialValues={
      Object {
        "email": "Sincere@april.biz",
        "name": "Leanne Graham",
        "phone": "1-770-736-8031 x56442",
        "username": "Bret",
        "website": "hildegard.org",
      }
    }
    onSubmit={[Function]}
  />
</Styled(Component)>
`;
